fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/derby/DerbyResourceReferenceDAO.java

```diff
@@ -157,6 +191,29 @@ public void doCanonicalValuesUpsert(String paramList, Collection<String> urls) {
     @Override
     protected void doCommonTokenValuesUpsert(String paramList, Collection<CommonTokenValue> tokenValues) {
 
+        insertToCommonTokenValuesTmp(tokenValues);
+
+        // Upsert the values from the declared global temp table into common_token_values
+        // ORDER BY helps to minimize the chance of deadlocks
+        StringBuilder upsert = new StringBuilder();
+        upsert.append("INSERT INTO common_token_values (token_value, code_system_id) ");
+        upsert.append("     SELECT src.token_value, src.code_system_id ");
+        upsert.append("       FROM SESSION.common_token_values_tmp src ");
+        upsert.append(" LEFT OUTER JOIN common_token_values ctv ");
+        upsert.append("              ON ctv.token_value = src.token_value ");
+        upsert.append("             AND ctv.code_system_id = src.code_system_id ");
+        upsert.append("      WHERE ctv.token_value IS NULL ");
+        upsert.append("   ORDER BY src.token_value, src.code_system_id");
+
+        try (Statement s = getConnection().createStatement()) {
+            s.executeUpdate(upsert.toString());
+        } catch (SQLException x) {
+            logger.log(Level.SEVERE, upsert.toString(), x);
+            throw getTranslator().translate(x);
+        }
+    }
+
+    private void insertToCommonTokenValuesTmp(Collection<CommonTokenValue> tokenValues) {
         final String insert = "INSERT INTO SESSION.common_token_values_tmp(token_value, code_system_id) VALUES (?, ?)";
```

Comment:
I wasn't totally sure if it was safe to use the same temporary table name here for both doCommonTokenValuesUpsert and the new readCommonTokenValueIds.

In the latest CI run I'm seeing errors like the following:
```
[ERROR   ] INSERT INTO common_token_values (token_value, code_system_id)      SELECT src.token_value, src.code_system_id        FROM SESSION.common_token_values_tmp src  LEFT OUTER JOIN common_token_values ctv               ON ctv.token_value = src.token_value              AND ctv.code_system_id = src.code_system_id       WHERE ctv.token_value IS NULL    ORDER BY src.token_value, src.code_system_id
A lock could not be obtained within the time requested
```

So I'm thinking to have this shared helper accept the name for the temporary table as an argument and then ensure that the two different method pass a unique value.

Reply:
scratch that, I went with the simpler `WHERE (c.token_value = ? AND c.code_system_id = ?) OR (c.token_value = ? AND c.code_system_id = ?) OR ...` approach instead because it just felt safer to me and I don't want to spend too much time optimizing for derby